turborepo-tests/helpers/setup_example_test.sh

```diff
@@ -1,37 +1,77 @@
-
-THIS_DIR=$(dirname "${BASH_SOURCE[0]}")
-MONOREPO_ROOT_DIR="$THIS_DIR/../.."
-TURBOREPO_TESTS_DIR="${MONOREPO_ROOT_DIR}/turborepo-tests"
-
-TARGET_DIR="$(pwd)"
-
-"${TURBOREPO_TESTS_DIR}/helpers/copy_fixture.sh" "${TARGET_DIR}" "${FIXTURE_NAME}" "${MONOREPO_ROOT_DIR}/examples"
-"${TURBOREPO_TESTS_DIR}/helpers/setup_git.sh" "${TARGET_DIR}"
-"${TURBOREPO_TESTS_DIR}/helpers/setup_package_manager.sh" "${TARGET_DIR}" "$PACKAGE_MANAGER"
-"${TURBOREPO_TESTS_DIR}/helpers/install_deps.sh" "$PACKAGE_MANAGER"
-
-# Set the TURBO_BINARY_PATH env var. The examples themselves invoke the locally installed turbo,
-# but turbo has an internal feature that will look for this environment variable and use it if it's set.
-# This is our way of running a locally built turbo version in our examples/ instead of the version
-# that is installed in the example's node_modules.
-if [ "${OSTYPE}" == "msys" ]; then
-  EXT=".exe"
-else
-  EXT=""
+export TURBO_TELEMETRY_MESSAGE_DISABLED=1
+
+# Start by figuring out which example we're testing and its package manager
+example_path=$1
+package_manager=$2
+
+if [ -z "$example_path" ]; then
+  echo "No example path was provided"
+  exit 1
 fi
 
-export TURBO_TELEMETRY_MESSAGE_DISABLED=1
-export TURBO_BINARY_PATH=${MONOREPO_ROOT_DIR}/target/debug/turbo${EXT}
-
-# Undo the set -eo pipefail at the top of this script
-# This script is called with a leading ".", which means that it does not run
-# in a new child process, so the set -eo pipefail would affect the calling script.
-# Some of our tests actually assert non-zero exit codes, and we don't want to
-# abort the test in those cases. So we undo the set -eo pipefail here.
-set +eo pipefail
+if [ -z "$package_manager" ]; then
+  echo "No package manager was provided"
+  exit 1
+fi
+
+# Use the right command for each package manager
+if [ "$package_manager" == "npm" ]; then
+  package_manager_command="npm install"
+elif [ "$package_manager" == "pnpm" ]; then
+  package_manager_command="pnpm install"
+elif [ "$package_manager" == "yarn" ]; then
+  package_manager_command="yarn"
+fi
+
+# All examples implement these two tasks
+# and it's reasonable to assume that they will continue to do so
+turbo_command="turbo build lint"
+
+# Head into a temporary directory
+mkdir -p ../../examples-tests-tmp
+cd ../../examples-tests-tmp
+
+# Head into the examples tests directory
+rm -rf "../examples-tests-tmp/$example_path"
+# Remove node_modules so the copy is faster when working locally
+find "../examples/$example_path" -name "node_modules" -type d -prune | xargs rm -rf
+cp -r "../examples/$example_path" $example_path
+
+# Isolate from git's perspective
+rm -rf $example_path/.git
+../turborepo-tests/helpers/setup_git.sh $example_path
+
+cd $example_path
+
+# Let's also isolate from turbo's perspective
+# in case you've ran the tests before
+rm -rf .turbo/ node_modules/ || true
+
+# Make /tmp dir for writing dump logs
+mkdir -p ./tmp
+echo -e "$(cat .gitignore)" "\n/tmp/" >.gitignore
```

Comment:
you can append to `.gitignore` with `>>` instead of 

```suggestion
echo "/tmp/" >> .gitignore
```

Reply:
Bless my heart, truly.